pkg/oc/admin/prune/imageprune/worker.go

```diff
@@ -0,0 +1,349 @@
+	case *imagegraph.ImageNode:
+		return fmt.Sprintf("failed to delete Image %s: %v", t.Image.DockerImageReference, pf.Err)
+	case *imagegraph.ImageComponentNode:
+		detail := ""
+		if isn, ok := pf.Parent.(*imagegraph.ImageStreamNode); ok {
+			detail = " in repository " + getName(isn.ImageStream)
+		}
+		switch t.Type {
+		case imagegraph.ImageComponentTypeConfig:
+			return fmt.Sprintf("failed to delete image config link %s%s: %v", t.Component, detail, pf.Err)
+		case imagegraph.ImageComponentTypeLayer:
+			return fmt.Sprintf("failed to delete image layer link %s%s: %v", t.Component, detail, pf.Err)
+		case imagegraph.ImageComponentTypeManifest:
+			return fmt.Sprintf("failed to delete image manifest link %s%s: %v", t.Component, detail, pf.Err)
+		default:
+			return fmt.Sprintf("failed to delete %s%s: %v", t.String(), detail, pf.Err)
+		}
+	default:
+		return fmt.Sprintf("failed to delete %v: %v", t, pf.Err)
+	}
+}
+
+// JobResult is a result of job's processing.
+type JobResult struct {
+	Job       *Job
+	Deletions []Deletion
+	Failures  []Failure
+}
+
+func (jr *JobResult) update(deletions []Deletion, failures []Failure) *JobResult {
+	jr.Deletions = append(jr.Deletions, deletions...)
+	jr.Failures = append(jr.Failures, failures...)
+	return jr
+}
+
+// Worker knows how to prune image and its related components.
+type Worker interface {
+	// Run is supposed to be run as a go-rutine. It terminates when nil is received through the in channel.
+	Run(in <-chan *Job, out chan<- JobResult)
+}
+
+type worker struct {
+	algorithm       pruneAlgorithm
+	registryClient  *http.Client
+	registryURL     *url.URL
+	imagePruner     ImageDeleter
+	streamPruner    ImageStreamDeleter
+	layerLinkPruner LayerLinkDeleter
+	blobPruner      BlobDeleter
+	manifestPruner  ManifestDeleter
+}
+
+var _ Worker = &worker{}
+
+// NewWorker creates a new pruning worker.
+// TODO: accept deleter factories rather than deleters themselves
+func NewWorker(
+	algorithm pruneAlgorithm,
+	registryClientFactory RegistryClientFactoryFunc,
+	registryURL *url.URL,
+	imagePruner ImageDeleter,
+	streamPruner ImageStreamDeleter,
+	layerLinkPruner LayerLinkDeleter,
+	blobPruner BlobDeleter,
+	manifestPruner ManifestDeleter,
+) (Worker, error) {
+	client, err := registryClientFactory()
+	if err != nil {
+		return nil, err
+	}
+
+	return &worker{
+		algorithm:       algorithm,
+		registryClient:  client,
+		registryURL:     registryURL,
+		imagePruner:     imagePruner,
+		streamPruner:    streamPruner,
+		layerLinkPruner: layerLinkPruner,
+		blobPruner:      blobPruner,
+		manifestPruner:  manifestPruner,
+	}, nil
+}
+
+func (w *worker) Run(in <-chan *Job, out chan<- JobResult) {
+	for {
+		job := <-in
+		if job == nil {
+			return
+		}
+		out <- *w.prune(job)
+	}
+}
+
+func (w *worker) prune(job *Job) *JobResult {
+	res := &JobResult{Job: job}
+
+	// If namespace is specified prune only ImageStreams and nothing more.
+	if len(w.algorithm.namespace) > 0 {
```

Comment:
When may this condition become true?

Reply:
Good catch, the statement was misplaced. It's removed now.